src/Algebra/Module/Morphism/Linear/Properties.agda

```diff
@@ -0,0 +1,206 @@
+-- The Agda standard library
+--
+-- Properties of linear maps.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra                   using (CommutativeRing)
+open import Algebra.Module            using (Module)
+open import Level                     using (Level; _⊔_)
+
+module Algebra.Module.Morphism.Linear.Properties
+  {r ℓr m ℓm : Level}
+  {ring      : CommutativeRing r ℓr}
+  (modA modB  : Module ring m ℓm)
+  where
+
+import Algebra.Module.Properties          as Properties
+import Function.Definitions
+import Relation.Binary.Reasoning.Setoid   as Reasoning
+import Algebra.Module.Morphism.Structures as MorphismStructures
+
+open import Agda.Builtin.Sigma
+open import Axiom.DoubleNegationElimination
+open import Data.List
+open import Data.Product
+  using (Σ; _,_; ∃; Σ-syntax; ∃-syntax; _×_)
+open import Relation.Nullary          using (¬_)
+open import Relation.Nullary.Negation using (contraposition)
+
+open MorphismStructures.ModuleMorphisms modA modB
+open Module modA
+  using () renaming
+  ( Carrierᴹ       to A
+  ; _+ᴹ_           to _+ᴬ_
+  ; _*ₗ_           to _·ᴬ_
+  ; -ᴹ_            to -ᴬ_
+  ; _≈ᴹ_           to _≈ᴬ_
+  ; 0ᴹ             to 0ᴬ
+  ; +ᴹ-comm        to +ᴬ-comm
+  ; +ᴹ-congˡ       to +ᴬ-congˡ
+  ; +ᴹ-congʳ       to +ᴬ-congʳ
+  ; *ₗ-zeroˡ       to ·ᴬ-zeroˡ
+  ; -ᴹ‿cong        to -ᴬ‿cong
+  ; -ᴹ‿inverseʳ    to -ᴬ‿inverseʳ
+  ; -ᴹ‿involutive  to -ᴬ‿involutive
+  ; uniqueʳ‿-ᴹ     to uniqueʳ‿-ᴬ
+  ; ≈ᴹ-setoid      to ≈ᴬ-setoid
+  ; ≈ᴹ-sym         to symᴬ
+  ; leftSemimodule to leftSemimoduleᴬ
+  )
+open Properties leftSemimoduleᴬ
+  using () renaming
+  ( non-zeroʳ to non-zeroʳᴬ
+  ; non-zeroˡ to non-zeroˡᴬ
+  )
+open Module modB
+  using () renaming
+  ( Carrierᴹ       to B
+  ; _+ᴹ_           to _+ᴮ_
+  ; _*ₗ_           to _·ᴮ_
+  ; -ᴹ_            to -ᴮ_
+  ; _≈ᴹ_           to _≈ᴮ_
+  ; 0ᴹ             to 0ᴮ
+  ; +ᴹ-comm        to +ᴮ-comm
+  ; +ᴹ-congˡ       to +ᴮ-congˡ
+  ; +ᴹ-congʳ       to +ᴮ-congʳ
+  ; *ₗ-zeroˡ       to ·ᴮ-zeroˡ
+  ; -ᴹ‿cong        to -ᴮ‿cong
+  ; -ᴹ‿inverseʳ    to -ᴮ‿inverseʳ
+  ; -ᴹ‿involutive  to -ᴮ‿involutive
+  ; uniqueʳ‿-ᴹ     to uniqueʳ‿-ᴮ
+  ; ≈ᴹ-setoid      to ≈ᴮ-setoid
+  ; ≈ᴹ-sym         to symᴮ
+  ; leftSemimodule to leftSemimoduleᴮ
+  )
+open Properties leftSemimoduleᴮ
+  using () renaming
+  ( non-zeroʳ to non-zeroʳᴮ
+  ; non-zeroˡ to non-zeroˡᴮ
+  )
+open CommutativeRing ring
+  using (_≈_; _*_; _+_) renaming
+  ( Carrier to S
+  ; 0#      to 𝟘
+  ; 1#      to 𝟙
+  )
+open module Reasoningᴬ = Reasoning ≈ᴬ-setoid
+  using () renaming
+  ( begin_ to beginᴬ_
+  ; _∎     to _∎ᴬ
+  )
+infixr 2 step-≈ᴬ
+step-≈ᴬ = Reasoningᴬ.step-≈
+syntax step-≈ᴬ x y≈z x≈y = x ≈ᴬ⟨ x≈y ⟩ y≈z
+open module Reasoningᴮ = Reasoning ≈ᴮ-setoid
+open Function.Definitions _≈ᴬ_ _≈ᴮ_
+
+_≉ᴬ_ : A → A → Set ℓm
+x ≉ᴬ y = ¬ (x ≈ᴬ y)
+
+_≉ᴮ_ : B → B → Set ℓm
+x ≉ᴮ y = ¬ (x ≈ᴮ y)
+
+module _
```

Comment:
One of the things you could do is to open reasoning in `modA` locally to this module, for example.

Reply:
Okay, let me give that a try.
Thanks for the tip! :)

Reply:
@JacquesCarette , can I open one or the other for the `module _` in general and then singly override my choice, using a `where open ...` that is local to a particular proof?
When I try this, Agda tells me that I've created ambiguity around the symbol: `begin_`; so, I'm guessing the answer is, "no."

I'm guessing the only alternative is to use a proof-local `where open ...` clause in _every_ proof that uses the equational reasoning symbols?

Reply:
You can't override what's visible in a name space. 

Note if you open things in a `private` block, that visibility won't leak out to the top-level (which is what happens with`_` modules). Sometimes I name my modules, and open them later, to also deal with visibility issues.

`where open ...` in every proof is always an option. But it does get noisy.

Reply:
Okay, I went with per-proof `where open ...` and it looks fine to my eye.

**These changes included in latest push and ready for review.**